pkg/endpoint/log.go

```diff
@@ -75,16 +78,16 @@ func (e *Endpoint) updateLogger() {
 		baseLogger.SetLevel(logrus.DebugLevel)
 	}
 
-	e.loggerMutex.Lock()
-	// Note: endpoint.loggerMutex protects the reference but not the logger objects. We
-	// cannot update the old object directly as that could be racey.
-	e.logger = baseLogger.WithFields(logrus.Fields{
+	l := baseLogger.WithFields(logrus.Fields{
```

Comment:
Don't we need to serialize calls to `updateLogger()`? Does it matter?

Reply:
I don't think it matters TBH

Reply:
@tgraf here